internal/benthos/benthos-builder/builders/sql.go

```diff
@@ -148,6 +152,14 @@ func (b *sqlSyncBuilder) BuildSourceConfigs(ctx context.Context, params *bb_inte
 		return nil, fmt.Errorf("unable to build select queries: %w", err)
 	}
 
+	for table, querymap := range tableRunTypeQueryMap {
+		for runtype, q := range querymap {
+			if runtype == tabledependency.RunTypeInsert {
```

Comment:
not sure I understand. this map is tracking whether a tables query is fk subset safe. the t/f value comes from the querybuilder

Reply:
so in this case youre only using the value if it is an insert, is that what's going on here?

Reply:
yeah I mean I could separate it by runtype but the isNotForeignKeySafeSubsetMap value is the same for insert or update. I could remove the runtype check and just have it overwrite the value since the insert and update are the same

Reply:
Idk, maybe just a nice comment would suffice